shop/src/main/java/ro/msg/learning/shop/DTOs/orderDto/SimpleProductQuantity.java

```diff
@@ -0,0 +1,23 @@
+package ro.msg.learning.shop.DTOs.orderDto;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class SimpleProductQuantity {
+    private Integer productId;
+    private Integer productQuantity;
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof SimpleProductQuantity)
+            return this.productId.equals(((SimpleProductQuantity) obj).productId) && this.productQuantity <= ((SimpleProductQuantity) obj).productQuantity;
```

Comment:
this.productQuantity "<= ". Use compareTo(Object obj) from Comparable interface.
@Data generates all the boilerplate that is normally associated with simple POJOs (Plain Old Java Objects) and beans: getters for all fields, setters for all non-final fields, and appropriate toString, equals and hashCode implementations that involve the fields of the class, and a constructor that initializes all final fields, as well as all non-final fields with no initializer that have been marked with @NonNull, in order to ensure the field is never null.